### PR TITLE
accessibility - large font size implemented in ListTransferMethod and…

### PR DESCRIPTION
**Pull request by asenthil-altmtrk**

### CODE REVIEW

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift [hunk 1]
```diff
@@ -31,9 +31,9 @@ public extension HyperwalletTransferMethodType {
         if let fees = self.fees?.nodes {
             let feeLabel = "add_transfer_method_fee_label".localized()
 
-            attributedText.append(value: feeLabel, font: Theme.Label.captionOne, color: color)
+            attributedText.append(value: feeLabel, font: UIFont.preferredFont(forTextStyle: .caption1), color: color)
             attributedText.append(value: HyperwalletFee.format(fees: fees),
-                                  font: Theme.Label.captionOneMedium,
+                                  font: UIFont.preferredFont(forTextStyle: .caption1),
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: This should still be Theme.Label.captionOneMedium. Update Theme class to assign ``` UIFont.preferredFont(forTextStyle: .caption1) ``` to the Theme.Label.captionOneMedium

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift [hunk 2]
```diff
@@ -31,9 +31,9 @@ public extension HyperwalletTransferMethodType {
         if let fees = self.fees?.nodes {
             let feeLabel = "add_transfer_method_fee_label".localized()
 
-            attributedText.append(value: feeLabel, font: Theme.Label.captionOne, color: color)
+            attributedText.append(value: feeLabel, font: UIFont.preferredFont(forTextStyle: .caption1), color: color)
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: same as below

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift [hunk 3]
```diff
@@ -48,8 +48,8 @@ public extension HyperwalletTransferMethodType {
                                          processingTimeLabel,
                                          "add_transfer_method_processing_time_label".localized())
 
-            attributedText.append(value: processingTimeLabel, font: Theme.Label.captionOne, color: color)
-            attributedText.append(value: processingTimeValue, font: Theme.Label.captionOneMedium, color: color)
+            attributedText.append(value: processingTimeLabel, font: UIFont.preferredFont(forTextStyle: .caption1), color: color)
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: same as above. Any changes made should be made in Theme class

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift [hunk 4]
```diff
@@ -49,7 +49,7 @@ public extension HyperwalletTransferMethodType {
                                          "add_transfer_method_processing_time_label".localized())
 
             attributedText.append(value: processingTimeLabel, font: Theme.Label.captionOne, color: color)
-            attributedText.append(value: processingTimeValue, font: Theme.Label.captionOneMedium, color: color)
+            attributedText.append(value: processingTimeValue, font: UIFont.preferredFont(forTextStyle: .caption1).semibold, color: color)
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: @whersbach-hw confirmed that since we are facing issues with large text accessibility for captionOneMedium we can use captionOne instead everywhere. So remove captionOneMedium from everywhere and set captionOne. Also pass the font and color to this method ``` func formatFeesProcessingTime(font: UIFont, color: UIColor) -> NSAttributedString ``` and use passed font and color. Also delete SelectTransferMethodTypeConfiguration and modify ``` func configure(configuration: SelectTransferMethodTypeConfiguration?) ``` to ``` func configure(configuration: HyperwalletTransferMethodType?) ``` and delete ``` func getCellConfiguration(indexPath: IndexPath) -> SelectTransferMethodTypeConfiguration ``` from the SelectTransferMethodTypePresenter

File: Common/Sources/Extensions/UIFont.swift [hunk 5]
```diff
@@ -20,6 +20,10 @@ import UIKit
 
 /// The UIFont extension
 public extension UIFont {
+    var bold: UIFont { return withWeight(.bold) }
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/UIFont.swift]: No need to add this

File: Common/Sources/Extensions/UIFont.swift [hunk 6]
```diff
@@ -45,4 +49,19 @@ public extension UIFont {
             return
         }
     }
+
+    /// To add weight to preferredFont
+    ///
+    /// - Parameters:
+    /// - weight : UIFont.Weight
+    private func withWeight(_ weight: UIFont.Weight) -> UIFont {
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/UIFont.swift]: remove this method as well

File: TransferMethod/Sources/AddTransferMethodController.swift [hunk 7]
```diff
@@ -71,7 +70,7 @@ public final class AddTransferMethodController: UITableViewController {
     private lazy var infoView: UIStackView = {
         let label = UILabel()
         label.numberOfLines = 0
-
+        label.adjustsFontForContentSizeCategory = true
```
Comment (jsaini-hw) [inline TransferMethod/Sources/AddTransferMethodController.swift]: set font and color to createAccountButton

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift (hunk 1 above)
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: Modified

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift (hunk 2 above)
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: Modified

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift (hunk 3 above)
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: Removed preferredFont

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift (hunk 4 above)
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: Changes done

File: Common/Sources/Extensions/UIFont.swift (hunk 5 above)
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/UIFont.swift]: Removed Weight extension

File: Common/Sources/Extensions/UIFont.swift (hunk 6 above)
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/UIFont.swift]: Removed Weight extension

File: TransferMethod/Sources/AddTransferMethodController.swift (hunk 7 above)
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/AddTransferMethodController.swift]: Font and textcolor added

File: Common/Sources/Extensions/HyperwalletTransferMethodType.swift [hunk 8]
```diff
@@ -24,17 +24,16 @@ public extension HyperwalletTransferMethodType {
     /// Formats the fee processing time
     ///
     /// - Returns: an NSAttributedString
-    func formatFeesProcessingTime() -> NSAttributedString {
+    func formatFeesProcessingTime(font: UIFont, color: UIColor) -> NSAttributedString {
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/HyperwalletTransferMethodType.swift]: Comments updated

File: TransferMethod/Sources/AddTransferMethodController.swift [hunk 9]
```diff
@@ -52,12 +52,11 @@ public final class AddTransferMethodController: UITableViewController {
...
         let button = UIButton()
         button.translatesAutoresizingMaskIntoConstraints = false
-        button.heightAnchor.constraint(
-            greaterThanOrEqualToConstant: Theme.Cell.smallHeight).isActive = true
-
         button.accessibilityLabel = "create_account_label".localized()
         button.accessibilityIdentifier = "createAccountButton"
         button.setTitle("create_account_label".localized(), for: .normal)
+        button.titleLabel?.adjustsFontForContentSizeCategory = true
+        button.titleLabel?.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/AddTransferMethodController.swift]: Changed to bodyfont

File: TransferMethod/Sources/CountryCurrencyCell.swift [hunk 10]
```diff
@@ -52,12 +52,17 @@ final class CountryCurrencyCell: GenericCell<GenericCellConfiguration> {
             accessibilityIdentifier = configuration.identifier
 
             titleLabel.text = configuration.title
+            titleLabel.lineBreakMode = .byWordWrapping
+            titleLabel.numberOfLines = 0
+            titleLabel.adjustsFontForContentSizeCategory = true
             titleLabel.accessibilityLabel = configuration.title
             titleLabel.accessibilityIdentifier = configuration.title
+            titleLabel.font = UIFont.preferredFont(forTextStyle: .caption1)
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/CountryCurrencyCell.swift]: Appdelegate updated with Thememanager

File: TransferMethod/Sources/CountryCurrencyCell.swift [hunk 11]
```diff
@@ -52,12 +52,17 @@ final class CountryCurrencyCell: GenericCell<GenericCellConfiguration> {
...
 
             valueLabel.text = accessoryType == .checkmark ? "" : configuration.value
             valueLabel.accessibilityLabel = configuration.value
             valueLabel.accessibilityIdentifier = configuration.value
+            valueLabel.font = UIFont.preferredFont(forTextStyle: .caption1)
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/CountryCurrencyCell.swift]: Updated

File: TransferMethod/Sources/ListTransferMethodCell.swift [hunk 12]
```diff
@@ -74,18 +73,28 @@ extension ListTransferMethodCell {
     /// - Parameter transferMethod: a transfer method which contains the info needs to be filled to the cell.
     public func configure(transferMethod: HyperwalletTransferMethod) {
         textLabel?.text = transferMethod.type?.lowercased().localized()
+        textLabel?.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/ListTransferMethodCell.swift]: Deleted

File: TransferMethod/Sources/ListTransferMethodCell.swift [hunk 13]
```diff
@@ -74,18 +73,28 @@ extension ListTransferMethodCell {
...
+        textLabel?.adjustsFontForContentSizeCategory = true
+        textLabel?.numberOfLines = 0
+        textLabel?.lineBreakMode = .byWordWrapping
         textLabel?.accessibilityIdentifier = "ListTransferMethodTableViewCellTextLabel"
         detailTextLabel?.attributedText = formatDetails(
             transferMethodCountry: transferMethod.transferMethodCountry?.localized() ?? "",
             additionalInfo: transferMethod.value)
-
+        detailTextLabel?.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/ListTransferMethodCell.swift]: Deleted

File: TransferMethod/Sources/ListTransferMethodCell.swift [hunk 14]
```diff
@@ -74,18 +73,28 @@ extension ListTransferMethodCell {
...
+        detailTextLabel?.adjustsFontForContentSizeCategory = true
         detailTextLabel?.numberOfLines = 0
         detailTextLabel?.lineBreakMode = .byWordWrapping
         let icon = UIImage.fontIcon(HyperwalletIcon.of(transferMethod.type ?? "").rawValue,
                                     Theme.Icon.frame,
                                     CGFloat(Theme.Icon.size),
                                     Theme.Icon.primaryColor)
+        if #available(iOS 11.0, *) {
+            imageView?.adjustsImageSizeForAccessibilityContentSizeCategory = true
+        } else {
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/ListTransferMethodCell.swift]: Removed else part

File: TransferMethod/Sources/SelectTransferMethodTypeCell.swift [hunk 15]
```diff
@@ -80,13 +80,24 @@ extension SelectTransferMethodTypeCell {
...
-            detailTextLabel?.attributedText = configuration.feesProcessingTime
+            detailTextLabel?.adjustsFontForContentSizeCategory = true
+
             let image = UIImage.fontIcon(configuration.transferMethodIconFont,
                                          Theme.Icon.frame,
                                          CGFloat(Theme.Icon.size),
                                          Theme.Icon.primaryColor)
+            if #available(iOS 11.0, *) {
+                imageView?.adjustsImageSizeForAccessibilityContentSizeCategory = true
+            } else {
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/SelectTransferMethodTypeCell.swift]: Removed else

File: TransferMethod/Sources/SelectTransferMethodTypePresenter.swift [hunk 16]
```diff
@@ -73,7 +73,8 @@ final class SelectTransferMethodTypePresenter {
     /// Return the `SelectTransferMethodTypeConfiguration` based on the index
     func getCellConfiguration(indexPath: IndexPath) -> SelectTransferMethodTypeConfiguration {
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/SelectTransferMethodTypePresenter.swift]: Removed this method

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 17]
```diff
@@ -30,15 +30,15 @@ final class ReceiptTransactionCell: UITableViewCell {
 
     lazy var receiptTypeLabel: UILabel = {
         let label = UILabel(frame: .zero)
-        label.font = Theme.Label.bodyFontMedium
+        label.font = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: This should move to ThemeManager

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 18]
```diff
@@ -30,15 +30,15 @@ final class ReceiptTransactionCell: UITableViewCell {
...
         label.textColor = Theme.Label.color
         label.accessibilityIdentifier = "receiptTransactionTypeLabel"
         return label
     }()
 
     lazy var amountLabel: UILabel = {
         let label = UILabel(frame: .zero)
-        label.font = Theme.Label.bodyFontMedium
+        label.font = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: same as above

File: Transfer/Sources/Extensions/TransferThemeManager.swift [hunk 19]
```diff
@@ -55,29 +55,29 @@ public extension ThemeManager {
...
         let proxy = TransferAllFundsCell.appearance()
-        proxy.titleLabelFont = Theme.Label.bodyFontMedium
+        proxy.titleLabelFont = Theme.Label.bodyFont
         proxy.titleLabelColor = Theme.Label.color
     }
 
     private static func applyToTransferAmountCell() {
         let proxy = TransferAmountCell.appearance()
-        proxy.titleLabelFont = Theme.Label.bodyFontMedium
+        proxy.titleLabelFont = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Transfer/Sources/Extensions/TransferThemeManager.swift]: Why did you change from bodyFontMedium to captionOne? Change it to bodyFont

File: Transfer/Sources/Extensions/TransferThemeManager.swift [hunk 20]
```diff
@@ -55,29 +55,29 @@ public extension ThemeManager {
...
         proxy.titleLabelColor = Theme.Label.color
-        proxy.currencyLabelFont = Theme.Label.bodyFontMedium
+        proxy.currencyLabelFont = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Transfer/Sources/Extensions/TransferThemeManager.swift]: Why did you change from bodyFontMedium to captionOne? Change it to bodyFont

File: Transfer/Sources/TransferAllFundsCell.swift [hunk 21]
```diff
@@ -88,6 +88,10 @@ final class TransferAllFundsCell: UITableViewCell {
     }
 
     func configure(setOn: Bool, _ handler: @escaping TransferAllFundsSwitchHandler) {
+        titleLabel.adjustsFontForContentSizeCategory = true
+        titleLabel.numberOfLines = 0
+        titleLabel.font = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Transfer/Sources/TransferAllFundsCell.swift]: This is being set from ThemeManager so remove it from here.

Comment (jsaini-hw) [reply Transfer/Sources/TransferAllFundsCell.swift]: Also update ``` @objc dynamic var titleLabelFont: UIFont! {
        get { return titleLabel.font }
        set {
            titleLabel.font = newValue
            amountTextField.font = newValue
        }
    }

    @objc dynamic var titleLabelColor: UIColor! {
        get { return titleLabel.textColor }
        set {
            titleLabel.textColor = newValue
            amountTextField.textColor = newValue
        }
    } ```

File: Transfer/Sources/TransferAmountCell.swift [hunk 22]
```diff
@@ -100,9 +100,19 @@ final class TransferAmountCell: UITableViewCell {
 
     func configure(amount: String?, currency: String?, isEnabled: Bool, _ handler: @escaping EnteredAmountHandler) {
         titleLabel.text = "transfer_amount".localized()
+        titleLabel.adjustsFontForContentSizeCategory = true
+        titleLabel.numberOfLines = 0
+        titleLabel.font = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Transfer/Sources/TransferAmountCell.swift]: This is being set from ThemeManager so remove it from here.
Also update `
@objc dynamic var titleLabelFont: UIFont! { 
   get { return titleLabel.font } 
   set { 
       titleLabel.font = newValue 
       amountTextField.font = newValue 
    } 
}
`
```@objc dynamic var titleLabelColor: UIColor! {
    get { return titleLabel.textColor }
    set {
        titleLabel.textColor = newValue
        amountTextField.textColor = newValue
    }
} ```

File: Transfer/Sources/TransferAmountCell.swift [hunk 23]
```diff
@@ -100,9 +100,19 @@ final class TransferAmountCell: UITableViewCell {
...
+        titleLabel.lineBreakMode = .byWordWrapping
         amountTextField.text = amount
         amountTextField.isEnabled = isEnabled
+        amountTextField.font = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Transfer/Sources/TransferAmountCell.swift]: This is being set from ThemeManager so remove it from here.

File: Transfer/Sources/TransferDestinationCell.swift [hunk 24]
```diff
@@ -112,18 +112,25 @@ final class TransferDestinationCell: GenericCell<HyperwalletTransferMethod> {
 extension TransferDestinationCell {
     func configure(transferMethod: HyperwalletTransferMethod) {
         titleLabel.text = transferMethod.title
+        titleLabel.font = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Transfer/Sources/TransferDestinationCell.swift]: This is being set from ThemeManager so remove it from here. As I have said so many times earlier, these values are set through ThemeManager and not directly.

File: Transfer/Sources/TransferDestinationCell.swift [hunk 25]
```diff
@@ -112,18 +112,25 @@ final class TransferDestinationCell: GenericCell<HyperwalletTransferMethod> {
...
+        titleLabel.adjustsFontForContentSizeCategory = true
         subtitleLabel.attributedText = formatDetails(
             subtitle: transferMethod.transferMethodCountry?.localized() ?? "",
             additionalInfo: transferMethod.value)
 
         subtitleLabel.numberOfLines = 0
+        subtitleLabel.font = Theme.Label.captionOne
```
Comment (jsaini-hw) [inline Transfer/Sources/TransferDestinationCell.swift]: same as above

File: TransferMethod/Sources/SelectTransferMethodTypeCell.swift [hunk 26]
```diff
@@ -75,18 +67,29 @@ final class SelectTransferMethodTypeCell: UITableViewCell {
 }
 
 extension SelectTransferMethodTypeCell {
-    func configure(configuration: SelectTransferMethodTypeConfiguration?) {
+    func configure(configuration: HyperwalletTransferMethodType?) {
         if let configuration = configuration {
-            accessibilityIdentifier = configuration.transferMethodTypeName
+            accessibilityIdentifier = configuration.name
+
+            textLabel?.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/SelectTransferMethodTypeCell.swift]: Removed Theme.Label.captionone

File: TransferMethod/Sources/SelectTransferMethodTypeCell.swift [hunk 27]
```diff
@@ -75,18 +67,29 @@ final class SelectTransferMethodTypeCell: UITableViewCell {
...
+            textLabel?.text = configuration.name
+            textLabel?.numberOfLines = 0
+            textLabel?.lineBreakMode = .byWordWrapping
+            textLabel?.adjustsFontForContentSizeCategory = true
 
-            textLabel?.text = configuration.transferMethodTypeName
+            detailTextLabel?.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/SelectTransferMethodTypeCell.swift]: Removed

File: TransferMethod/Sources/SelectTransferMethodTypeCell.swift [hunk 28]
```diff
@@ -75,18 +67,29 @@ final class SelectTransferMethodTypeCell: UITableViewCell {
...
+            detailTextLabel?.attributedText = configuration
+                .formatFeesProcessingTime(font: Theme.Label.captionOne, color: Theme.Label.subTitleColor)
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/SelectTransferMethodTypeCell.swift]: Updated

File: TransferMethod/Sources/SelectTransferMethodTypeController.swift [hunk 29]
```diff
@@ -112,10 +116,6 @@ extension SelectTransferMethodTypeController {
         presenter.navigateToAddTransferMethod(indexPath.row)
         tableView.deselectRow(at: indexPath, animated: true)
     }
-
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/SelectTransferMethodTypeController.swift]: Modified

File: Demo/AppDelegate.swift [hunk 30]
```diff
@@ -30,6 +30,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Override point for customization after application launch.
 
         //ThemeManager.applyWhiteTheme()
+        ThemeManager.applyTransferMethodTheme()
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Demo/AppDelegate.swift]: Added

File: Transfer/Sources/Extensions/TransferThemeManager.swift (hunk 19 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/Extensions/TransferThemeManager.swift]: Changed

File: Transfer/Sources/Extensions/TransferThemeManager.swift (hunk 20 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/Extensions/TransferThemeManager.swift]: Changed

File: Transfer/Sources/TransferAllFundsCell.swift (hunk 21 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/TransferAllFundsCell.swift]: Removed

File: Transfer/Sources/TransferAmountCell.swift (hunk 22 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/TransferAmountCell.swift]: Changed made

File: Transfer/Sources/TransferDestinationCell.swift (hunk 24 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/TransferDestinationCell.swift]: Removed

File: Transfer/Sources/TransferDestinationCell.swift (hunk 25 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/TransferDestinationCell.swift]: Removed

File: TransferMethod/Sources/CountryCurrencyCell.swift [hunk 31]
```diff
@@ -51,14 +51,16 @@ final class CountryCurrencyCell: GenericCell<GenericCellConfiguration> {
             }
             accessibilityIdentifier = configuration.identifier
 
+            titleLabel.font = Theme.Label.bodyFont
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/CountryCurrencyCell.swift]: Removed

File: TransferMethod/Sources/CountryCurrencyCell.swift [hunk 32]
```diff
@@ -51,14 +51,16 @@ final class CountryCurrencyCell: GenericCell<GenericCellConfiguration> {
...
             titleLabel.text = configuration.title
+            titleLabel.adjustsFontForContentSizeCategory = true
             titleLabel.accessibilityLabel = configuration.title
             titleLabel.accessibilityIdentifier = configuration.title
-
             valueLabel.text = accessoryType == .checkmark ? "" : configuration.value
+            valueLabel.adjustsFontForContentSizeCategory = true
+            valueLabel.font = Theme.Label.bodyFont
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/CountryCurrencyCell.swift]: Removed

File: TransferMethod/Sources/ListTransferMethodCell.swift [hunk 33]
```diff
@@ -74,18 +73,26 @@ extension ListTransferMethodCell {
     /// - Parameter transferMethod: a transfer method which contains the info needs to be filled to the cell.
     public func configure(transferMethod: HyperwalletTransferMethod) {
         textLabel?.text = transferMethod.type?.lowercased().localized()
+        textLabel?.adjustsFontForContentSizeCategory = true
+        textLabel?.numberOfLines = 0
+        textLabel?.lineBreakMode = .byWordWrapping
         textLabel?.accessibilityIdentifier = "ListTransferMethodTableViewCellTextLabel"
+        textLabel?.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/ListTransferMethodCell.swift]: Removed

File: TransferMethod/Sources/ListTransferMethodCell.swift [hunk 34]
```diff
@@ -74,18 +73,26 @@ extension ListTransferMethodCell {
...
         detailTextLabel?.attributedText = formatDetails(
             transferMethodCountry: transferMethod.transferMethodCountry?.localized() ?? "",
             additionalInfo: transferMethod.value)
-
         detailTextLabel?.accessibilityIdentifier = "ListTransferMethodTableViewCellDetailTextLabel"
+        detailTextLabel?.adjustsFontForContentSizeCategory = true
         detailTextLabel?.numberOfLines = 0
+        detailTextLabel?.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/ListTransferMethodCell.swift]: Removed

File: Transfer/Sources/CreateTransferController.swift [hunk 35]
```diff
@@ -107,6 +107,9 @@ extension CreateTransferController {
             withIdentifier: TransferTableViewFooterView.reuseIdentifier) as? TransferTableViewFooterView else {
                 return nil
         }
+        view.footerLabel.adjustsFontForContentSizeCategory = true
+        view.footerLabel.font = Theme.Label.captionOne
```
[the remark this answers is not in the document]
Comment (asenthil-altmtrk) [reply Transfer/Sources/CreateTransferController.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 17 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: Moved to ThemeManager

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 18 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: Moved to ThemeManager

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 36]
```diff
@@ -143,7 +136,9 @@ extension ReceiptTransactionCell {
         let iconFont = HyperwalletIcon.of(entry).rawValue
 
         receiptTypeLabel.text = receipt.type?.rawValue.lowercased().localized()
-
+        receiptTypeLabel.adjustsFontForContentSizeCategory = true
+        receiptTypeLabel.numberOfLines = 0
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 37]
```diff
@@ -143,7 +136,9 @@ extension ReceiptTransactionCell {
...
+        receiptTypeLabel.lineBreakMode = .byWordWrapping
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 38]
```diff
@@ -152,9 +147,18 @@ extension ReceiptTransactionCell {
             ? Theme.Amount.creditColor
             : Theme.Amount.debitColor
 
+        amountLabel.adjustsFontForContentSizeCategory = true
+        amountLabel.numberOfLines = 0
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 39]
```diff
@@ -152,9 +147,18 @@ extension ReceiptTransactionCell {
...
+        amountLabel.lineBreakMode = .byWordWrapping
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 40]
```diff
@@ -152,9 +147,18 @@ extension ReceiptTransactionCell {
...
         createdOnLabel.text = formattedCreatedOn
         currencyLabel.text = receipt.currency
 
+        createdOnLabel.adjustsFontForContentSizeCategory = true
+        createdOnLabel.numberOfLines = 0
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 41]
```diff
@@ -152,9 +147,18 @@ extension ReceiptTransactionCell {
...
+        createdOnLabel.lineBreakMode = .byWordWrapping
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 42]
```diff
@@ -152,9 +147,18 @@ extension ReceiptTransactionCell {
...
+        currencyLabel.adjustsFontForContentSizeCategory = true
+        currencyLabel.numberOfLines = 0
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift [hunk 43]
```diff
@@ -152,9 +147,18 @@ extension ReceiptTransactionCell {
...
+        currencyLabel.lineBreakMode = .byWordWrapping
```
Comment (jsaini-hw) [inline Receipt/Sources/ReceiptTransactionCell.swift]: remove this

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 36 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 37 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 38 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 39 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 40 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 41 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 42 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Receipt/Sources/ReceiptTransactionCell.swift (hunk 43 above)
Comment (asenthil-altmtrk) [reply Receipt/Sources/ReceiptTransactionCell.swift]: removed

File: Common/Sources/Extensions/UIFont.swift [hunk 44]
```diff
@@ -45,4 +45,19 @@ public extension UIFont {
             return
         }
     }
+
+    /// Indicates if current preferred content size category belongs to large or not
+    static var isLargeSizeCategory: Bool {
+        return self.largeSizes.contains(UIApplication.shared.preferredContentSizeCategory)
+    }
+
+    static var largeSizes: [UIContentSizeCategory] {
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/UIFont.swift]: Make it private since it's not being used from anywhere else

Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/UIFont.swift]: Changed to private
